### PR TITLE
dependency updates

### PR DESCRIPTION
**Pull request by jakobbraun**

### CODE REVIEW

File: doc/changes/changes_2.4.0.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# BucketFS Java 2.4.0, released 2022-??-??
+
+Code name:
+
+## Summary
+
+## Features
+
+* ISSUE_NUMBER: description
```
Comment (redcatbear) [inline doc/changes/changes_2.4.0.md:9]: issue number missing for update

**Review by redcatbear — DISMISSED** [review]

File: .github/workflows/ci-build.yml [hunk 2]
```diff
@@ -8,48 +9,65 @@ on:
...
       cancel-in-progress: true
     steps:
       - name: Checkout the repository
-        uses: actions/checkout@v3
+        uses: actions/checkout@v2
         with:
           fetch-depth: 0
       - name: Set up JDK 11
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v2
```
Comment (kaklakariada) [inline .github/workflows/ci-build.yml:27]:
```suggestion
        uses: actions/setup-java@v3
```

File: .github/workflows/ci-build.yml [hunk 3]
```diff
@@ -8,48 +9,65 @@ on:
...
-          cache: 'maven'
-      - name: Cache SonarCloud packages
-        uses: actions/cache@v3
+      - name: Configure Testcontainers
+        run: |
+          echo '{}' > "$HOME"/.docker/config.json
+          echo 'docker.client.strategy=org.testcontainers.dockerclient.EnvironmentAndSystemPropertyClientProviderStrategy' > "$HOME"/.testcontainers.properties
+          echo 'testcontainers.reuse.enable=true' >> "$HOME"/.testcontainers.properties
+      - name: Cache local Maven repository
+        uses: actions/cache@v2
```
Comment (kaklakariada) [inline .github/workflows/ci-build.yml:37]:
```suggestion
        uses: actions/cache@v3
```

File: .github/workflows/ci-build.yml [hunk 4]
```diff
@@ -8,48 +9,65 @@ on:
...
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - name: Cache SonarQube packages
+        uses: actions/cache@v2
```
Comment (kaklakariada) [inline .github/workflows/ci-build.yml:44]:
```suggestion
        uses: actions/cache@v3
```

File: .github/workflows/ci-build.yml [hunk 5]
```diff
@@ -8,48 +9,65 @@ on:
...
-        if: ${{ always() && github.event.pull_request.head.repo.full_name == github.repository && github.actor != 'dependabot[bot]' }}
-        with:
-          github_token: ${{ secrets.GITHUB_TOKEN }}
+          mvn --batch-mode --update-snapshots clean verify \
+              -DtrimStackTrace=false \
+              -Dtest.coverage="true" \
+              -Dcom.exasol.dockerdb.image=${{ matrix.docker_db_version }}
       - name: Sonar analysis
-        if: ${{ env.SONAR_TOKEN != null }}
+        if: ${{ env.DEFAULT_DB_VERSION == matrix.docker_db_version }}
```
Comment (kaklakariada) [inline .github/workflows/ci-build.yml:56]:
```suggestion
        if: ${{ env.SONAR_TOKEN != null && env.DEFAULT_DB_VERSION == matrix.docker_db_version }}
```

**Review by kaklakariada — CHANGES_REQUESTED** [review]

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**